mne_bids/write.py

```diff
@@ -1070,6 +1096,56 @@ def write_anat(bids_root, subject, t1w, session=None, acquisition=None,
              'NAS': list(mri_landmarks[1, :]),
              'RPA': list(mri_landmarks[2, :])}
         fname = t1w_basename.replace('.nii.gz', '.json')
+        if op.isfile(fname) and not overwrite:
+            raise IOError('Wanted to write a file but it already exists and '
+                          '`overwrite` is set to False. File: "{}"'
+                          .format(fname))
         _write_json(fname, t1w_json, overwrite, verbose)
 
+        if deface:
+            # x: L/R L+, y: S/I I+, z: A/P A+
+            t1w_data = t1w.get_data().copy()
+            idxs_vox = np.meshgrid(np.arange(t1w_data.shape[0]),
+                                   np.arange(t1w_data.shape[1]),
+                                   np.arange(t1w_data.shape[2]),
+                                   indexing='ij')
+            idxs_vox = np.array(idxs_vox)  # (3, *t1w_data.shape)
+            idxs_vox = np.transpose(idxs_vox,
+                                    [1, 2, 3, 0])  # (*t1w_data.shape, 3)
+            idxs_vox = idxs_vox.reshape(-1, 3)  # (n_voxels, 3)
+
+            mri_landmarks_ras = apply_trans(t1w.affine, mri_landmarks)
+            ras_meg_t = \
+                get_ras_to_neuromag_trans(*mri_landmarks_ras[[1, 0, 2]])
+
+            idxs_ras = apply_trans(t1w.affine, idxs_vox)
+            idxs_meg = apply_trans(ras_meg_t, idxs_ras)
+
+            # now comes the actual defacing
+            # 1. move center of voxels to (nasion - inset)
+            # 2. rotate the head by theta from the normal to the plane passing
+            # through anatomical coordinates
+            trans_y = -mri_landmarks_ras[1, 1] + t1w_data.shape[2] * inset
+            idxs_meg = apply_trans(translation(y=trans_y), idxs_meg)
+            idxs_meg = apply_trans(rotation(x=-np.deg2rad(theta)), idxs_meg)
+            coords = idxs_meg.reshape(t1w.shape + (3,))  # (*t1w_data.shape, 3)
+            mask = (coords[..., 2] < 0)   # z < 0
+
+            t1w_data[mask] = 0.
+            # smooth decided against for potential lack of anonymizaton
+            # https://gist.github.com/alexrockhill/15043928b716a432db3a84a050b241ae
+
+            t1w = nib.Nifti1Image(t1w_data, t1w.affine, t1w.header)
```

Comment:
finally can you make this code go in a private function that would be in charge of all the parameter validation.